types/fined/index.d.ts

```diff
@@ -7,11 +7,15 @@ export = fined;
 
 declare function fined(path: string | fined.PathSpec, opts?: fined.PathSpec): fined.Result | null;
 
+interface StringObject {
+  toString(): string;
+}
+
 declare namespace fined {
     interface PathSpec {
         path?: string | undefined;
         name?: string | undefined;
-        extensions?: string | string[] | { [extension: string]: string | null } | undefined;
+        extensions?: string | string[] | { [extension: string]: string | null } | StringObject | StringObject[] | undefined;
```

Comment:
Did you mean to include strings constructed via a `String` constructor (via `new String()`)? Then the type to use is not your custom `StringObject` but simply `String` (uppercase `S`).

And if you're introducing this, then please take into account that the `path` parameter/property is also allowed to be a `String`, as are the parameters `cwd` and `name`.